### PR TITLE
RWA-976: Handling of error scenarios for inserting messages in the da…

### PR DESCRIPTION
**Pull request by rasikasri**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-976

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by sullivangareth — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/CcdCaseEventsDeadLetterQueueConsumer.java [hunk 1]
```diff
@@ -39,24 +39,32 @@ public void run() {
...
+                            messageId
+                        );
+                    } catch (Exception ex) {
+                        log.error("Error processing CCD Case Event Dead Letter Queue message with id '{}' - "
+                                      + "abandon the processing and ASB will re-deliver it", messageId);
+                        receiver.abandon(message);
+                    }
+                });
+        } catch (Exception ex) {
+            log.error("Error occurred while closing the session", ex);
```
Comment (sullivangareth) [inline src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/CcdCaseEventsDeadLetterQueueConsumer.java:67]: This receiver being used is not session enabled so this log message should be changed

File: src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/CcdCaseEventConsumerTest.java [hunk 2]
```diff
@@ -82,11 +82,29 @@ void given_session_is_accepted_when_receiver_complete_throws_error_on_both_calls
...
 
-        verify(receiverClient, Mockito.times(2)).complete(receivedMessage);
-        verify(receiverClient, Mockito.times(0)).abandon(any());
+        verify(receiverClient, Mockito.times(1)).complete(receivedMessage);
+        verify(receiverClient, Mockito.times(1)).abandon(receivedMessage);
+        verify(receiverClient, Mockito.times(0)).deadLetter(any(), any());
+    }
+
+    @Test
+    void given_session_is_accepted_when_handling_message_throws_error() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/CcdCaseEventConsumerTest.java:97]: can we rename these test to follow the should_  pattern? the given syntax does not read completely clearly as there is not then to qualify the outcome?

**Review by BenNzewi — COMMENTED** [review]
Can we cover these in the Integration test?

Given we're consuming a message from the DLQ
When we can't obtain a connection to the database
Then abort the processing of the message as ASB will re-deliver it
 

Given we're consuming a message from the DLQ
And we have obtained a connection to the DB
When an Exception is thrown during insertion
Then abort the processing of the message as ASB will re-deliver it
 

Given we've inserted a message in the DB
When we're encountering an error while calling complete
Then abort the processing of the message as ASB will re-deliver it 
And the insertion statement will update the record already present (not create a new one)

File: src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/CcdCaseEventConsumerTest.java (hunk 2 above)
Comment (rasikasri) [reply src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/CcdCaseEventConsumerTest.java:97]: All the ASB service test classes follow the patter of given_session_is

**Pull request merged**